src/AppContainer.tsx

```diff
@@ -0,0 +1,322 @@
+      const updatedData: Todo = {
+        id,
+        userId: USER_ID,
+        title: query.trim(),
+        completed: !completed,
+      };
+
+      await updateTodo(id, updatedData);
+      setTodos((state) => state.map((todo) => {
+        if (todo.id === id) {
+          return {
+            ...todo,
+            completed: !completed,
+          };
+        }
+
+        return todo;
+      }));
+      setIsUpdatingId((state) => state.filter((stateId) => stateId !== id));
+    } catch {
+      handleErrorMessage(ErrorMessage.updateTodo);
+      setIsUpdatingId((state) => state.filter((stateId) => stateId !== id));
+    }
+  };
+
+  const toggleAll = async () => {
+    const isStatus = todos.every((todo) => todo.completed);
+
+    try {
+      const updatedTodos = await Promise.all(
+        todos.map(async (todo) => {
+          if ((isStatus && todo.completed) || (!isStatus && !todo.completed)) {
+            setIsUpdatingId((state) => [...state, todo.id]);
+            const updatedData: Todo = {
+              id: todo.id,
+              userId: USER_ID,
+              title: query.trim(),
+              completed: !isStatus,
+            };
+
+            await updateTodo(todo.id, updatedData);
+
+            return { ...todo, completed: !isStatus };
+          }
+
+          return todo;
+        }),
+      );
+
+      setTodos(updatedTodos);
+      setIsUpdatingId([]);
+    } catch {
+      handleErrorMessage(ErrorMessage.updateTodo);
+      setIsUpdatingId([]);
+    }
+  };
+
+  const handleDelete = async (id?: number) => {
+    if (id) {
+      try {
+        setIsUpdatingId(state => (
+          [...state, id]
+        ));
+        await deleteTodo(id);
+        setTodos((state) => {
+          return [...state.filter(todo => todo.id !== id)];
+        });
+        setIsUpdatingId((state) => (
+          [...state.filter(stateId => stateId !== id)]
+        ));
+      } catch {
+        handleErrorMessage(ErrorMessage.deleteTodo);
+        setIsUpdatingId((state) => (
+          [...state.filter(stateId => stateId !== id)]
+        ));
+      }
+    } else {
+      setTodos(stateMain => {
+        stateMain.forEach(async (todo) => {
+          try {
+            if (todo.completed) {
+              setIsUpdatingId(state => (
+                [...state, todo.id]
+              ));
+              await deleteTodo(todo.id);
+              setTodos((state) => {
+                return [...state.filter(todoF => todoF.id !== todo.id)];
+              });
+              setIsUpdatingId((state) => (
+                [...state.filter(stateId => stateId !== todo.id)]
+              ));
+            }
+          } catch {
+            handleErrorMessage(ErrorMessage.deleteTodo);
+            setIsUpdatingId([]);
+          }
+        });
+
+        return stateMain;
+      });
+    }
+  };
+
+  const ecsKeyCancel = (e: string) => {
+    if (e === 'Escape') {
+      setEditQuery(editQueryPrev);
+      setIsEditing(null);
+    }
+  };
```

Comment:
I like to invert such a conditions
```suggestion
  const ecsKeyCancel = (e: string) => {
    if (e !== 'Escape') return;
    
    setEditQuery(editQueryPrev);
    setIsEditing(null);
  };
```